### PR TITLE
Feature/second layer iteration 2

### PR DESCRIPTION
**Pull request by paulusrex**
## Description
This is a first approach to second layer. It's functional but doesn't have the styles properly setted and needs to be completed with the behaviour for legitimateIntereses ~= consent

## Solves ticket/s
https://jira.scmspain.com/browse/PSP-3287

## Expected behavior
Show purposes, specialPurposes,  features, and specialFeatures. Vendors are shown too but maybe need to be moved to another page.

## Review steps
Use the demo or the demo of tcf/ui linking tcf/secondLayer

Screenshot 2020-07-08 at 16 25 11

## Further considerations
Any NB features could be addressed in [3rd iteration](https://jira.scmspain.com/browse/PSP-3308)

TODO:
- Remove? the vendors from this layer to a new link
- Add logic to use legitimateInterest and Consent in a proper way
- Styles (pending some UX designs)
- Style LegalExpandedContent as MarkDown or similar
- Extract from hardcoding the  ADEVINTA_COLLECTED_CONSENTS to be injected by configuration 

## Memetized description
![layers](https://media.giphy.com/media/j5QFLjGfXKvCs7jmHC/giphy.gif

### CODE REVIEW

File: components/tcf/secondLayer/src/settings.js [hunk 1]
```diff
@@ -1,10 +1,23 @@
+export const ADEVINTA_COLLECTED_CONSENTS = {
```
Comment (alextremp) [inline components/tcf/secondLayer/src/settings.js:1]: ❗ 
remember to open a tech debt task for this, as it's not mentioned in __further considerations__

File: demo/tcf/utils/borosMock.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ const mockedEmptyUserConsent = {
 }
 const mockedFullUserConsent = {
   isNew: false,
-  specialFeatureOptins: [],
```
Comment (alextremp) [inline demo/tcf/utils/borosMock.js:16]: is this OK?
the TCData key is `specialFeatureOptins` (and is not an array)

Comment (paulusrex) [reply demo/tcf/utils/borosMock.js:16]: This is not the TCData is our domain object

File: components/tcf/secondLayer/src/components/tcf-secondLayer-legal-expandedContent/index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export default function TcfSecondLayerLegalExpandedContent({baseClass, info}) {
+  return (
+    <>
+      <h4>{info.description}</h4>
```
Comment (maryammdot) [inline components/tcf/secondLayer/src/components/tcf-secondLayer-legal-expandedContent/index.js:7]: You are not using classes to control css here?

**Pull request merged**